src/tools/compiletest/src/header.rs

```diff
@@ -574,12 +574,14 @@ impl Config {
         if line.starts_with(prefix) && line.as_bytes().get(prefix.len()) == Some(&b'-') {
             let name = line[prefix.len()+1 ..].split(&[':', ' '][..]).next().unwrap();
 
+            let triple: Vec<&str> = self.target.split('-').collect();
+
             name == "test" ||
-                util::matches_os(&self.target, name) ||             // target
-                name == util::get_arch(&self.target) ||             // architecture
-                name == util::get_pointer_width(&self.target) ||    // pointer width
-                name == self.stage_id.split('-').next().unwrap() || // stage
-                Some(name) == util::get_env(&self.target) ||        // env
+                util::matches_os(&self.target, &triple, name) || // target
+                name == util::get_arch(&triple) ||               // architecture
+                name == util::get_pointer_width(&self.target) ||       // pointer width
+                name == self.stage_id.split('-').next().unwrap() ||    // stage
+                Some(&name) == triple.get(3) ||                        // env
```

Comment:
Nit: This is a pure refactoring, right? If so, I'd prefer to leave it out of this PR, for consistency. (Splitting multiple times is sure to have completely neglegible performance impact.)